packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -149,7 +150,11 @@ export class AwsS3Publish implements PublisherBase {
           Body: fileContent,
         };
 
-        uploadPromises.push(this.storageClient.putObject(params));
+        // Rate limit the concurrent execution of file uploads to batches of 10
+        const uploadFile = limiter(async () =>
```

Comment:
you can skip the async here (doesn't hurt to have it but still)

Reply:
think it would make it clearer to make the changes in both and just have it as `limiter(client.putObject)`; shouldn't use a language feature if it's not directly applicable 🤔

Reply:
no that actually doesn't work - `limiter(client.putObject)` is different from `limiter(() => client.putObject)` and you'll have to make sure to use the latter :)

Reply:
hm that makes sense. I assumed calling it directly would work since the argument is a promise-return/async fn 🤔 

I guess calling it directly would then not actually rate limit anything? In this way we defer the execution to the limiter? (if my question makes sense 😅 )